tests/test_core.py

```
@@ -593,6 +687,62 @@ def test_api_scenario_generation(self):
+        test_resources_dir = os.path.join(
+            resources_dir(), "scenario_generation", "data_path_deprecation"
+        )
+
+        # Use the test runner that only compares the scenarios
+        default_runner = kh.get_runner()
+        test_runner = ScenarioWriterRunner(self, test_resources_dir)
+        kh.set_runner(test_runner)
+
+        # Obtain mock arguments for each API call
+        method_test_args = self._build_mock_deprecated_data_path_api_method_parameters()
+
+        # Test for each dataset mock parameters
+        for method_name, method_full_args in method_test_args.items():
+            # Set the runners test name
+            test_runner.test_name = method_name
+
+            # Clean the directory for this method's tests
+            cleanup_dir(test_runner.output_scenario_dir, "*/output/*._kh", verbose=True)
+            test_runner.subtest_name = "Customer"
+            with self.subTest(method=method_name):
+                # Get the API function and its args and kwargs
+                method = getattr(kh, method_name)
+                args = method_full_args["args"]
+                kwargs = method_full_args["kwargs"]
+
+                # Test that using legacy paths entails a deprecation warning
+                with warnings.catch_warnings(record=True) as warning_list:
+                    method(*args, **kwargs)
+
+                # Check the warning message
+                self.assertTrue(len(warning_list) > 0)
+                for warning in warning_list:
+                    self.assertTrue(issubclass(warning.category, UserWarning))
+                    warning_message = warning.message
+                    self.assertEqual(len(warning_message.args), 1)
+                    message = warning_message.args[0]
+                    self.assertTrue(
+                        "'`'-based dictionary data path" in message
+                        and "deprecated" in message
+                    )
```

I'd modify this a bit, because this fails if there are warnings that we do not control: Instead of checking that every warning matches the one you expect, find just one that does it.

It could be something like having a "warning_found" boolean that you set to `True` when you find a warning that matches. Then after the loop you `assertTrue(warning_found)`.

---

Done.

---

You have to eliminate the assertions inside the loop if not it will continue to fail.

The inner loop should be something like:
```python
for ... :
    deprecation_warning_found = (
        issubclass(warning.category, UserWarning) and 
        len(warning_message.args) == 1 and 
        "'`'-based dictionary data path" in message and 
        "deprecated" in message`
    )
    if deprecation_warning_found:
        break
self.assertTrue(deprecation_warning_found)
```
We loose a bit of "assert granularity" but it is better to have a test that fails only when the expected situation does not appear.

---

Yes, right, we want to check all 3 conditions anyway, and we cannot assume that any warning satisfies 2 conditions (except for the message itself).
So, OK for the change.

---

Done.